examples/exoskeleton/js/views/app-view.js

```diff
@@ -8,6 +8,15 @@ var app = app || {};
 		el.style.display = toggle ? '' : 'none';
 	};
 
+	var matchesSelector = function (el, selector) {
+		var node = el;
+		var nodes = (node.parentNode || node.document).querySelectorAll(selector);
+		var i = -1;
+
+		while (nodes[++i] && nodes[i] !== node);
+		return !!nodes[i];
```

Comment:
It's a neat hack, but we're trying to be as approachable as possible for newcomers. The functional alternatives are usually a bit more readable. What about this?

```js
[].slice.call(document.querySelectorAll("span")).some(function (el) { return el === node; })
```